reflection.md

```diff
@@ -1,12 +1,15 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+Implementing regular expressions using a DSL involves defining a set of operators that can be used to construct regular language expressions in a more natural and intuitive way. In this implementation, the easiest operators to implement were the basic operators such as concatenation, union and the Kleene star, which have well-defined and widely accepted semantics. These operators are also relatively straightforward to implement using pattern matching, recursion and other standard programming techniques.
```

Comment:
Wooooo Scala is great. idk if you have taken programming languages, but scala is so much nice then haskell (imo)